# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/chromium/src/mojo/core/ports/event.h
# Commit: 7b7819250f79
# Full Hash: 7b7819250f7965227783233af6de0e4e1b789af6
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2021-06-23 09:53:24
# Regressor Bug: 1706374
# File Overlap Count: 1
# Description:
#   Bug 1706374 - Part 11: Add NodeController component bridging IPC and Ports, r=handyman
#   
#   The NodeController and NodeChannel types act as the backbone connecting the
#   existing IPC logic and driving the ports routing code. Individual NodeChannel
#   objects wrap and respond to messages from IPC::Channel, and the NodeController
# ==============================================================================

diff -r 8bdac5eedf78 -r 7b7819250f79 ipc/chromium/src/mojo/core/ports/event.h
--- a/ipc/chromium/src/mojo/core/ports/event.h	Tue Jun 22 18:17:21 2021 +0000
+++ b/ipc/chromium/src/mojo/core/ports/event.h	Tue Jun 22 18:17:22 2021 +0000
@@ -130,6 +130,13 @@
     return static_cast<const T*>(message_.get());
   }
 
+  template <typename T>
+  mozilla::UniquePtr<T> TakeMessage() {
+    DCHECK(HasMessage());
+    DCHECK_EQ(&T::kUserMessageTypeInfo, message_->type_info());
+    return mozilla::UniquePtr<T>{static_cast<T*>(message_.release())};
+  }
+
   void ReservePorts(size_t num_ports);
   bool NotifyWillBeRoutedExternally();
 